### PR TITLE
Support passwords stored as SCRAM-SHA-256

### PR DESCRIPTION
**Pull request by jfleury-gorgias**
This allows passing the password as plain text, a md5 hash or a SCRAM-SHA-256 hash. The password will be compared to whatever hash is stored in Postgres and updated accordingly.

The SCRAM-SHA-256 hash algorithm for password storage was introduced in Postgres version 10 and made the default in Postgres version 14.

### CODE REVIEW

**Comment by jfleury-gorgias** [comment]
I applied the recommendation from the previous PR (using `hmac.compare_digest` instead of `==`) and tested on staging.

Here are the logs before/after:

```
# Before
DEBUG:attributes.py:analyze_attributes:69 - Starting to analyze role [SNIPPED]
DEBUG:attributes.py:__init__:108 - self.rolename set to [SNIPPED]
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolbypassrls": "False"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolcanlogin": "True"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolconnlimit": "-1"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolcreatedb": "False"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolcreaterole": "False"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolinherit": "True"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolpassword": "SCRAM-SHA-256$4096:[SNIPPED]"
DEBUG:attributes.py:set_all_attributes:214 - Altering password for role "[SNIPPED]"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolreplication": "False"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolsuper": "False"
DEBUG:attributes.py:get_attribute_value:193 - Returning attribute "rolvaliduntil": "None"

# After
DEBUG:attributes.py:analyze_attributes:69 - Starting to analyze role [SNIPPED]
DEBUG:attributes.py:__init__:118 - self.rolename set to [SNIPPED]
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolbypassrls": "False"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolcanlogin": "True"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolconnlimit": "-1"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolcreatedb": "False"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolcreaterole": "False"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolinherit": "True"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolpassword": "SCRAM-SHA-256$4096:[SNIPPED]"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolreplication": "False"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolsuper": "False"
DEBUG:attributes.py:get_attribute_value:203 - Returning attribute "rolvaliduntil": "None"
```

See the line `DEBUG:attributes.py:set_all_attributes:214 - Altering password for role "[SNIPPED]"` that appears before my changes (because pgbedrock was comparing md5 to SCRAM password).

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # We only use Python for coveralls; both Python 2.7 and 3.6 are
```
Comment (FlipEnergy) [inline .travis.yml]: comment for python 3.7 here. Though not that it really matters since no travis CI is running here.

**Review by JordanP — APPROVED** [review]
LGTM.

**Comment by JordanP** [comment]
@johanfleury @FlipEnergy let's not forget about this one :)

**Pull request merged**